### DISCUSSION TITLE
user assignRole failed while import excel with upsert

### DISCUSSION THREAD

**Discussion by irfansbom** [spatie/laravel-permission#2154] [Q&A] [answered]
I tried to import users using laravel excel form maatwebsite with implement Withupserts and the uniqeBy() function using  'email',  then directly assignRole from that excel but it doesn't assign well. i tried to change using 
$user = new User();
$user->name = $row[1];
$user->email = $row[3];
$user->save();
$user->assignRole($row[5]);
it return SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry

then I change it to 
$user =  new User([
'name' =>  $row[1],
'email' => $row[3]
]);
$user->assignRole($row[5]);
it run succesfully  and replace the name but nothing happened to the role

**Comment by angeljqv** [comment]
Make a debug

**Reply by irfansbom** [reply]
I debug it and goes well, actually the assignRole run well whille not implement WithUpserts on importing laravel excel

**Reply by angeljqv** [reply]
Don't wait many answers, really hard that somebody else uses maatwebsite with this package, also only run `with/without` is not a debug

**Reply by irfansbom** [reply]
I got the solution, impement WithUpsert are returning affectd row, so i need to call user back then assign it

**Comment by irfansbom** [comment, marked as answer]
I got the solution, impement WithUpsert are returning affectd row, so i need to call user back then assign it
